extensions/native/compiler/src/ir/instructions.rs

```diff
@@ -319,6 +319,15 @@ pub enum DslIr<C: Config> {
     CycleTrackerStart(String),
     /// End the cycle tracker used by a block of code annotated by the string input.
     CycleTrackerEnd(String),
+
+    /// Sumcheck calculate layer eval
+    SumcheckLayerEval(
+        Ptr<C::N>,          // Input ctx: round, num_prod_spec, num_logup_spec, num_variables
+        Ptr<C::N>,          // Challenges: alpha, coeffs
+        Ptr<C::N>,          // prod_specs_eval
+        Ptr<C::N>,          // logup_specs_eval
+        Ptr<C::N>           // output
```

Comment:
Nitpick: can you add comment to explain the layout of output? like 
1. for computing expected evaluation, `output = [\sum_i alpha^i * prod[i][0] * prod[i][1] + \sum_j alpha^(2j) * (logup_q[i][0] * logup_q[i][1] + alpha* logup_p[i][0] * logup_q[i][1] + alpha * logup_p[i][1] * logup_q[i][0]]`;